.github/workflows/typescript-build-and-test.yml

```diff
@@ -0,0 +1,25 @@
+name: build-and-test
+
+on:
+  workflow_call:
+
+jobs:
+  build_and_test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "lts/*"
+      - run: yarn install
+      - run: yarn build
```

Comment:
I would run the unit tests before trying to build. Why even build if the unit tests have the potential of failing?

Reply:
why try and run the unit tests if the code doesn't build?

Reply:
the only reason you can even run the unit tests before building is because the unit tests are actually building the typescript into javascript on the fly for you.  in any compiled language, you can't run code until you compile it.

i just haven't ever heard this argument before, and looking around at ci for typescript, i can't find anyone running unit tests before building.

is there something you are concerned about, or think is a downfall to building before testing?

Reply:
Nothing of serious concern. I was thinking if there is an issue with the unit tests, we can catch that before taking the time to build which in my experience has been faster than running a build. But I suppose since we are working with typescript it would have to transpile before running the tests anyways.

So, you can ignore me. I was just thinking about trying to fail fast(er) if possible 😄 .